lib/elixir/lib/system.ex

```diff
@@ -604,12 +604,21 @@ defmodule System do
   Variable names and their values are strings.
   """
   @spec get_env() :: %{optional(String.t()) => String.t()}
-  def get_env do
-    Enum.into(:os.getenv(), %{}, fn var ->
-      var = IO.chardata_to_string(var)
-      [k, v] = String.split(var, "=", parts: 2)
-      {k, v}
-    end)
+
+  if function_exported?(:os, :env, 0) do
```

Comment:
You want to do it inside the function, otherwise only people who precompiled Elixir on Erlang/OTP 24+ would get the benefits. :) Also please add a TODO:

    # TODO: Remove this once we require Erlang/OTP 24+

Reply:
I think this will trigger an error since we're using ` --warnings-as-errors`

Reply:
Oh, nvm. I found how to disable the compiler error :)